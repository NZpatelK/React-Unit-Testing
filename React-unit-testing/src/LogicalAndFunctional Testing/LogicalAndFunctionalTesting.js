import React, { useState } from 'react';  
import DisplayName from './DispalyName';  
export const validateName = (name) => {  
 if((name && isNaN(name)) || !name)  
   return false  
 return true  
}  
const User = () => {  
 const [name,setName] = useState(null);  
 const [error,setError] = useState(false);  
  const setUserName = (event) =>{  
   const {value} = event.target;  
   var isOk = validateName(value);  
   setError(isOk);  
   setName(value);  
 }  
 return (  
   <div>  
    <DisplayName name={name} />  
    <input type="text" value={name} onChange={setUserName}/>  
     {error &&<div>Error</div>}  
 </div>  
 );  
};  
export default User;  
