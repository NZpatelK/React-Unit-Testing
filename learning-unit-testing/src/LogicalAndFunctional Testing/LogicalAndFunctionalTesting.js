import React, { useState } from 'react';  
import DisplayName from './DispalyName';  

/**
 * This class to validate when user input their name to make their name is string and only character. 
 */


/**
 * This function is to validate if the input name is string or not. 
 * @param {input name} name 
 * @returns boolean
 */

export const validateName = (name) => {  
 if((name && isNaN(name)) || !name)  
   return false  
 return true  
}  

/**
 * This is input function and display in the screen to allow the user can type their name and display output of their name
 * @returns render the display with input function and output of their input name. 
 */

const User = () => {  
 const [name,setName] = useState(null);  
 const [error,setError] = useState(false);  
  const setUserName = (event) =>{  
   const {value} = event.target;  
   var isOk = validateName(value);  
   setError(isOk);  
   setName(value);  
 }  

 return (  
   <div>  
    <DisplayName name={name} />  
    <input type="text" value={name} onChange={setUserName}/>  
     {error &&<div>Error</div>}  
 </div>  
 );  
};  

export default User;  
