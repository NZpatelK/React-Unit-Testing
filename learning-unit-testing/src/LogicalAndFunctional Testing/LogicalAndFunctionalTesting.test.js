import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import DisplayName from './DispalyName';
import { validateName } from './LogicalAndFunctionalTesting';

//This is to test the text that should be in dsiplay.
describe('DisplayName', () => {  
    test('should render text', () => {  
      const component = renderer.create(<DisplayName name={"Patel"} />);  
      let tree = component.toJSON();  
      expect(tree).toMatchSnapshot();  
    });  
});   

describe('Validation', () => {
    test('can set the error if input is number', () => {
        const name = 123;
        const error = validateName(name);
        expect(error).toBeTruthy();
    });
    test('Could reset error if input is empty', () => {
        const name = null;
        const error = validateName(name);
        expect(error).toEqual(false);
    });
})  
