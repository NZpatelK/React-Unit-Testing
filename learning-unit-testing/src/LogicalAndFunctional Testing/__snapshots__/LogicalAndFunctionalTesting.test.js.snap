// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayName should render text 1`] = `
<div>
  Welcome 
  Patel
  !
</div>
`;
